### DISCUSSION TITLE
Can I make Lingui work with Nextjs (TS)

### DISCUSSION THREAD

**Discussion by JaccoGoris** [lingui/js-lingui#1287] [Q&A]
I've found a lot of resources and guides online, even found one from logrocket which helps you set up a nextjs project. in the inital steps they are working with .js and .jsx files and "it just works".

later the article magically switches to .tsx as if that also just works. This is definitely not the case. I've tried a bunch of things, even ejecting out of SWC by adding a custom babelrc (broke my project), but I couldn't make Linguijs understand typescript. This feels very strange for a modern library.

Has anyone gotten it to work? or a good example of this? Thanks!

**Comment by landsman** [comment]
Probably not:

- https://github.com/lingui/js-lingui/issues/1158
- https://github.com/lingui/js-lingui/issues/1224

**Comment by timofei-iatsenko** [comment]
https://github.com/lingui/swc-plugin/tree/main/examples/nextjs-13
